server/heroku/processes.go

```diff
@@ -100,23 +99,19 @@ func (h *PostProcess) ServeHTTPContext(ctx context.Context, w http.ResponseWrite
 	}
 
 	if form.Attach {
-		inStream, outStream, err := hijackServer(w)
-		if err != nil {
-			return err
+		stream := &hijack.HijackReadWriter{
+			Response: w,
+			Header:   "HTTP/1.1 200 OK\r\nContent-Type: application/vnd.empire.raw-stream\r\n\r\n",
 		}
-		defer closeStreams(inStream, outStream)
-
-		fmt.Fprintf(outStream, "HTTP/1.1 200 OK\r\nContent-Type: application/vnd.empire.raw-stream\r\n\r\n")
-
+		defer stream.Close()
 		// Prevent the ELB idle connection timeout to close the connection.
-		defer close(streamhttp.Heartbeat(outStream, 10*time.Second))
+		defer close(streamhttp.Heartbeat(stream, 10*time.Second))
```

Comment:
I think in reality, we'd almost never hit that case, because errors returned from Run would rarely take longer than 10 seconds.